libr/debug/p/native/linux/linux_debug.c

```diff
@@ -126,12 +130,11 @@ int linux_handle_signals (RDebug *dbg, int tid) {
 				} else {
 					dbg->reason.bp_addr = (ut64)(size_t)siginfo.si_addr;
 					dbg->reason.type = R_DEBUG_REASON_BREAKPOINT;
-					// Switch to the thread that hit the breakpoint
-					r_debug_select (dbg, dbg->pid, tid);
-					dbg->tid = tid;
 				}
 			}
-		}
+		} break;
```

Comment:
It can be inside as well. That `break` is to prevent it from falling through and running code below.

I usually see it outside, for example:

https://github.com/radareorg/radare2/blob/154416c8fd4ce0a399c42e27ba88a0c535e320a0/libr/core/cmd_debug.c#L4859-L4864